tests/test_traits.py

```diff
@@ -343,3 +348,70 @@ def test_filter_value_validation_filter_size_3():
             np.array([1, 2, 3, 4, 5, 6, 7, 8, 9], dtype=np.float64), 3
         ),
     )
+
+
+class NormalAccessorWidget(BaseLayer):
+    _rows_per_chunk = 2
+
+    table = pa.table({"data": [1, 2, 3]})
+
+    value = NormalAccessor()
+
+
+def test_normal_accessor_validation_list_length():
+    # tuple or list must have 3 elements
+    with pytest.raises(TraitError):
+        NormalAccessorWidget()
+
+    with pytest.raises(TraitError, match="expected 3 values if passed a tuple or list"):
+        NormalAccessorWidget(value=(1, 2))
+
+    with pytest.raises(TraitError, match="expected 3 values if passed a tuple or list"):
+        NormalAccessorWidget(value=(1, 2, 3, 4))
+    NormalAccessorWidget(value=(1, 2, 3))
+    NormalAccessorWidget(value=(1, 2, 3))
+
+
+def test_normal_accessor_validation_list_type():
+    # tuple or list must be of scalar type
+    with pytest.raises(TraitError, match="expected scalar values"):
+        NormalAccessorWidget(value=["1.1", 2.2, 3.3])
+    NormalAccessorWidget(value=[1.1, 2.2, 3.3])
+
+
+def test_normal_accessor_validation_dim_shape_np_arr():
+    # expected value is a numpy ndarray with two dimensions
+    # And  size of the second dimension must be 3 i.e. `(N,3)`
+    with pytest.raises(TraitError, match="expected shape (N,3)"):
+        NormalAccessorWidget(value=np.array([1, 2], 3))
```

Comment:
All of these numpy calls will fail.

To construct a 2d numpy array you can do either
```py
np.array([[1, 2], [3, 4]])
```
which gives a 2x2 numpy array or 
```
np.array([1, 2, 3, 4, 5, 6]).reshape(-1, 2)
```
which first creates a 1-dimensional array and then rearranges into a 2-d array, in this case of shape `(3, 2)`.

Reply:
Right. I'll bring this as a general testing question because I thought that we were intentionally creating fail conditions and ensuring that the `NormalAcc` fails on these objects (like where the shape is just incorrect).

Reply:
Yeah I mean that we want to test that passing a _valid_ numpy array into `NormalAccessorWidget` fails. If the numpy creation fails before passing into our code, we haven't tested our own code yet.